autogluon/utils/tabular/features/feature_metadata.py

```diff
@@ -0,0 +1,270 @@
+        for type in metadata.type_group_map_special:
+            metadata.type_group_map_special[type] = [rename_map.get(feature, feature) for feature in metadata.type_group_map_special[type]]
+        return metadata
+
+    # Joins two metadata objects together, returning a new metadata object
+    # TODO: Add documentation on shared_raw_features usage
+    def join_metadata(self, metadata, shared_raw_features='error'):
+        if shared_raw_features not in ['error', 'error_if_diff', 'overwrite']:
+            raise ValueError(f"shared_raw_features must be one of {['error', 'error_if_diff', 'overwrite']}, but was: '{shared_raw_features}'")
+        type_map_raw = copy.deepcopy(self.type_map_raw)
+        shared_features = []
+        shared_features_diff_types = []
+        for key, features in metadata.type_map_raw.items():
+            if key in type_map_raw:
+                shared_features.append(key)
+                if type_map_raw[key] != metadata.type_map_raw[key]:
+                    shared_features_diff_types.append(key)
+        if shared_features:
+            if shared_raw_features == 'error':
+                logger.error('ERROR: Conflicting metadata:')
+                logger.error('Metadata 1:')
+                self.print_feature_metadata_full(log_prefix='\t', log_level=40)
+                logger.error('Metadata 2:')
+                metadata.print_feature_metadata_full(log_prefix='\t', log_level=40)
+                raise AssertionError(f"Metadata objects to join share raw features, but `shared_raw_features='error'`. Shared features: {shared_features}")
+            if shared_features_diff_types:
+                if shared_raw_features == 'overwrite':
+                    logger.log(20, f'Overwriting type_map_raw during FeatureMetadata join. Shared features with conflicting types: {shared_features_diff_types}')
+                    shared_features = []
+                elif shared_raw_features == 'error_if_diff':
+                    logger.error('ERROR: Conflicting metadata:')
+                    logger.error('Metadata 1:')
+                    self.print_feature_metadata_full(log_prefix='\t', log_level=40)
+                    logger.error('Metadata 2:')
+                    metadata.print_feature_metadata_full(log_prefix='\t', log_level=40)
+                    raise AssertionError(f"Metadata objects to join share raw features but do not agree on raw dtypes, and `shared_raw_features='error_if_diff'`. Shared conflicting features: {shared_features_diff_types}")
+        type_map_raw.update({key: val for key, val in metadata.type_map_raw.items() if key not in shared_features})
+
+        type_group_map_special = self.add_type_group_map_special([self.type_group_map_special, metadata.type_group_map_special])
+
+        return FeatureMetadata(type_map_raw=type_map_raw, type_group_map_special=type_group_map_special)
+
+    @staticmethod
+    def add_type_group_map_special(type_group_map_special_lst: list):
```

Comment:
nit: types: list type + return types.

Also, this method is not used outside of this class, should it be private?

Reply:
Added + made private